apps/server/internal/workspace/materializer.go

```
@@ -0,0 +1,264 @@
+
+	ready, err := m.store.MarkWorkspaceBootstrapReady(ctx, project.ID, issue.ID, current.ID, finalPath, source, baseBranch, baseRevision, current.WorkingBranch)
+	if err != nil {
+		return store.Workspace{}, fmt.Errorf("mark workspace ready: %w", err)
+	}
+	return ready, nil
+}
+
+func (m *Materializer) recoverPublished(ctx context.Context, current store.Workspace, finalPath, source, baseBranch string) (store.Workspace, bool, error) {
+	info, err := os.Stat(finalPath)
+	if err != nil || !info.IsDir() || !m.git.IsRepository(ctx, finalPath) {
+		return store.Workspace{}, false, nil
```

golangci-lint (2.13.2)

[error] error is not nil (line 155) but it returns nil

(nilerr)